class10.1/FootballTournamentSimulator.cpp

```diff
@@ -0,0 +1,221 @@
+    }
+
+    
+    int getGoalsTeam1() const { return goalsTeam1; }
+    int getGoalsTeam2() const { return goalsTeam2; }
+    double getTopHeightTeam1() const { return topHeightTeam1; }
+    double getTopHeightTeam2() const { return topHeightTeam2; }
+
+    
+    const Team& getTeam1() const { return team1; }
+    const Team& getTeam2() const { return team2; }
+};
+
+
+class Tournament {
+private:
+    std::vector<Team> teams;
+    std::vector<Match> matches;
+    std::map<std::string, double> points;
+    std::map<std::string, int> goalsDelta;
+    std::map<std::string, int> goalsScored;
+    std::map<std::string, int> redCards;
+    std::map<std::string, int> yellowCards;
+    std::map<std::string, double> topHeights;
+
+public:
+    Tournament(const std::vector<std::string>& teamNames) {
+        for (const auto& name : teamNames) {
+            teams.emplace_back(name);
+            points[name] = 0.0;
+            goalsDelta[name] = 0;
+            goalsScored[name] = 0;
+            redCards[name] = 0;
+            yellowCards[name] = 0;
+            topHeights[name] = 0.0;
+        }
+       
+        for (size_t i = 0; i < teams.size(); ++i) {
+            for (size_t j = i + 1; j < teams.size(); ++j) {
+                matches.emplace_back(teams[i], teams[j]);
+            }
+        }
+    }
+
+    void playTournament() {
+       
+        for (auto& match : matches) {
+            match.playMatch();
+
+           
+            updateStatistics(match);
+        }
+
+        
+        calculatePoints();
+    }
+
+    void printResults() const {
+      
+        auto sortedTeams = teams;
+        std::ranges::sort(sortedTeams, [&](const Team& t1, const Team& t2) {
+            if (points.at(t1.getName()) != points.at(t2.getName())) {
+                return points.at(t1.getName()) > points.at(t2.getName());
+            } else if (goalsDelta.at(t1.getName()) != goalsDelta.at(t2.getName())) {
+                return goalsDelta.at(t1.getName()) > goalsDelta.at(t2.getName());
+            } else if (goalsScored.at(t1.getName()) != goalsScored.at(t2.getName())) {
+                return goalsScored.at(t1.getName()) > goalsScored.at(t2.getName());
+            } else if (redCards.at(t1.getName()) != redCards.at(t2.getName())) {
+                return redCards.at(t1.getName()) < redCards.at(t2.getName());
+            } else if (yellowCards.at(t1.getName()) != yellowCards.at(t2.getName())) {
+                return yellowCards.at(t1.getName()) < yellowCards.at(t2.getName());
+            } else {
+                return topHeights.at(t1.getName()) > topHeights.at(t2.getName());
+            }
+        });
+
+       
+        std::cout << "Final Tournament Results:" << std::endl;
+       
+        std::cout << "| Team Name | Points | Goals Delta | Goals Scored | Red Cards | Yellow Cards | Top Heights |" << std::endl;
+        
+        for (const auto& team : sortedTeams) {
+            std::cout << "| " << team.getName() << " | ";
```

Comment:
you can use std::println with custom format provided inside {} to guarantee expected layout (see https://www.cppstories.com/2023/fun-print-tables-format/ as a reference)